.github/workflows/esbuild-publish.yml

```diff
@@ -15,32 +20,20 @@ on:
         required: false
 
 jobs:
-  build:
+  bundle:
+    if: ${{ inputs.branch || (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success')}}
+    uses: ./.github/workflows/bundle.yml
+    secrets: inherit
+    with:
+      branch: ${{ inputs.branch || 'main' }}
+      nodeVersion: ${{ inputs.nodeVersion || 'lts/*'}}
+  publish:
+    needs: bundle
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
```

Comment:
I do not think the job of publish will get the results from bundle since they are not sharing the same context. Let's just keep esbuild-publish as it used to be and not reuse the bundle workflow you created within this workflow.

Reply:
I'm just confused about how it works in testCoreBundling and couldn't work here?